pkg/kubelet/dockertools/manager.go

```diff
@@ -1336,7 +1336,7 @@ func (dm *DockerManager) KillPod(pod *api.Pod, runningPod kubecontainer.Pod) err
 			glog.Errorf("Failed tearing down the infra container: %v", err)
 			errs <- err
 		}
-		if err := dm.KillContainerInPod(networkContainer.ID, networkSpec, pod); err != nil {
+		if err := dm.KillContainerInPod(networkContainer.ID, networkSpec, pod, "Need to kill pod."); err != nil {
```

Comment:
That makes sense. Can we make the ```reason``` a bit more descriptive noting that the pod is being killed and will [not] be restarted?